File: atlas-frontend/src/api/assignments/dto.ts
export interface AssignmentDTO {
  id: number,
  name: string,
  description: string,
  classroom_id: number,
  teacher_id: number,
  unit_id: number,
  posted: string,
  rubric_id: number,
  deadline: string,
  attachments: string[]
}

export interface AssignmentMessageDTO {
  id: number,
  content: string,
  assignment_id: number,
  sender_id: number,
  receiver_id: number
}

File: atlas-server/crates/api_routes/src/assignments/dto.rs
use atlas_db_schema::models::Assignment;
use serde::{Deserialize, Serialize};

#[derive(Deserialize)]
pub struct NewAssignmentDTO {
    pub classroom_id: i32,
    pub name: String,
    pub description: String,
    pub rubric_id: i32,
    pub unit_id: i32,
    pub attachments: Vec<String>,
    pub deadline: String
}

#[derive(Serialize)]
pub struct NewAssignmentMessage {
    pub message: String,
    pub assignment_id: i32
}

#[derive(Deserialize)]
pub struct GetAssignmentDTO {
    pub assignment_id: i32
}

#[derive(Serialize)]
pub struct GetAssignmentMessage {
    pub message: String,
    pub assignment: Assignment
}
